src/octopeer-github/main/Options/DoNotWatchOptions.ts

```diff
@@ -8,62 +9,108 @@
+
+    public shouldElementBeWatched(element: ElementID) {
+        return !this.getElements().contains(element);
+    }
+
+    public shouldEventBeWatched(event: EventID) {
+        return !this.getEvents().contains(event);
+    }
+
+    public shouldCombinationBeWatched(combination: ElementXEventID) {
+        return !this.getCombinations().contains(combination);
+    }
+
     /**
      * Gets Elements not to Log, from the chrome storage.
-     * @returns {ElementSelectionBehaviourData[]}
+     * @returns {ElementID[]}
      */
     public getElements() {
-        let doNotWatchElements: ElementSelectionBehaviourData[] = [];
-
-        if (Options.get(Options.DNW_COMMENT_ELEMENTS)) {
-            doNotWatchElements.push(esbFactory.findElementSelectionBehaviourData(ElementID.CONFIRM_INLINE_COMMENT));
-            doNotWatchElements.push(esbFactory.findElementSelectionBehaviourData(ElementID.CREATE_PR_COMMENT));
-            doNotWatchElements.push(esbFactory.findElementSelectionBehaviourData(ElementID.EDIT_COMMENT));
+        let doNotWatchElements: ElementID[] = [];
+        for (let tuple of this.elementsThatShouldNotBeWatched) {
+            if (Options.get(tuple[0])) {
+                for (let element of tuple[1]) {
+                    doNotWatchElements.push(element);
+                }
+            }
         }
         return doNotWatchElements;
     }
+
     /**
      * Gets Events not to Log, from the chrome storage.
-     * @returns {ElementEventBindingCreatable[]}
+     * @returns {EventID[]}
      */
     public getEvents() {
-        let doNotWatchEvents: ElementEventBindingData[] = [];
-
-        if (Options.get(Options.DNW_ON_SCREEN_EVENTS)) {
-            doNotWatchEvents.push(eebFactory.findElementEventBindingData(EventID.SCROLL_INTO_VIEW));
-            doNotWatchEvents.push(eebFactory.findElementEventBindingData(EventID.SCROLL_OUT_OF_VIEW));
-        }
-        if (Options.get(Options.DNW_HOVER_EVENTS)) {
-            doNotWatchEvents.push(eebFactory.findElementEventBindingData(EventID.MOUSE_ENTER));
-            doNotWatchEvents.push(eebFactory.findElementEventBindingData(EventID.MOUSE_LEAVE));
-        }
-        if (Options.get(Options.DNW_KEYBOARD_EVENTS)) {
-            doNotWatchEvents.push(eebFactory.findElementEventBindingData(EventID.KEYSTROKE));
+        let doNotWatchEvents: EventID[] = [];
+        for (let tuple of this.eventsThatShouldNotBeWatched) {
+            if (Options.get(tuple[0])) {
+                for (let event of tuple[1]) {
+                    doNotWatchEvents.push(event);
+                }
+            }
         }
         return doNotWatchEvents;
     }
 
     /**
      * Gets Combinations not to Log, from the chrome storage.
      * This is currently just a placeholder for the structure to be used.
-     * @returns {ElementXEventCreatable[]}
+     * @returns {ElementXEventID[]}
      */
     public getCombinations() {
```

Comment:
That's unfortunately not possible. To that end, the DNWOptions would need to have the `combinationsThatShouldNotBeWatched` should be filled with combinations that should indeed not be logged. This is not part of this issue. :)